nni/retiarii/experiment/pytorch.py

```diff
@@ -131,6 +137,13 @@ def start(self, port: int = 8080, debug: bool = False) -> None:
     def _create_dispatcher(self):
         return self._dispatcher
 
+    def _strategy_monitor(self):
+        while True:
+            time.sleep(2)
+            if not self._strategy_thread.is_alive():
+                self._dispatcher.mark_experiment_as_ending()
+                break
```

Comment:
why not directly use `self._strategy_thread.join()` here?

Reply:
Make sense to me.